diesel/src/expression_methods/eq_all.rs

```diff
@@ -29,14 +28,14 @@ macro_rules! impl_eq_all {
             ($($Left,)+): EqAll<($($Right,)+)>,
         {
             type Output = Grouped<And<
-                Nullable<<$Left1 as EqAll<$Right1>>::Output>,
+                <$Left1 as EqAll<$Right1>>::Output,
                 <($($Left,)+) as EqAll<($($Right,)+)>>::Output,
             >>;
 
             fn eq_all(self, rhs: ($Right1, $($Right,)+)) -> Self::Output {
                 let ($Left1, $($Left,)+) = self;
                 let ($Right1, $($Right,)+) = rhs;
-                $Left1.eq_all($Right1).and(($($Left,)+).eq_all(($($Right,)+)))
+                $Left1.eq_all($Right1).and::<_, Bool>(($($Left,)+).eq_all(($($Right,)+)))
```

Comment:
So this is the part that is not solved: apparently now compiler is not able to infer which `ST` to use there, which I suspect is because of [this change](https://github.com/diesel-rs/diesel/commit/7ff5e6327d5e88095cb4b342da13bf06d6d0d1cd#diff-1b2dc4703512cd6842e04129b8eab9628b1c555158c7a4cfa332f220b81e5da4) (tests are exploding in a lot of places for the same reason).

Now that probably means we'd have to revert this change, but then it looks like changes from https://github.com/diesel-rs/diesel/commit/7ff5e6327d5e88095cb4b342da13bf06d6d0d1cd#diff-1b2dc4703512cd6842e04129b8eab9628b1c555158c7a4cfa332f220b81e5da4 to https://github.com/diesel-rs/diesel/commit/7ff5e6327d5e88095cb4b342da13bf06d6d0d1cd#diff-04a3838f33b9f0c5cdb21bfaf94920c0bec33b778c02ce092c23ded8d97f2700 as well as https://github.com/diesel-rs/diesel/commit/7ff5e6327d5e88095cb4b342da13bf06d6d0d1cd#diff-7c94f0bbca60b1d1a8174a803d1699d766b1d7365598f296e0879a8f4a5764bb depend on that change, and I'm not sure what these imply and what reverting them would imply.

@weiznich can you clarify what these changes imply?

Reply:
Ok so next commit does revert these and seems to pass test suite.